From 1089cc8c9adf3229f0fbd32de539f2384d7646a3 Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Thu, 17 Jun 2021 13:45:12 +0200
Subject: [PATCH 2/2] tuxtxt-enigma2: try to guess return values

  fix compiler warnings
  The return values used here are hard to guess. Please review.

../git/tuxtxt/tuxtxt.c: In function 'tuxtxt_run_ui':
../git/tuxtxt/tuxtxt.c:267:3:
 warning: 'return' with no value, in function returning non-void
  267 |   return;
      |   ^~~~~~
../git/tuxtxt/tuxtxt.c:178:5:
 note: declared here
  178 | int tuxtxt_run_ui(int pid, int demux)
      |     ^~~~~~~~~~~~~
../git/tuxtxt/tuxtxt.c:277:3:
 warning: 'return' with no value, in function returning non-void
  277 |   return;
      |   ^~~~~~
../git/tuxtxt/tuxtxt.c:178:5:
 note: declared here
  178 | int tuxtxt_run_ui(int pid, int demux)
      |     ^~~~~~~~~~~~~

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 tuxtxt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tuxtxt.c b/tuxtxt.c
index f550688..9907420 100644
--- a/tuxtxt.c
+++ b/tuxtxt.c
@@ -264,7 +264,7 @@ int tuxtxt_run_ui(int pid, int demux)
 	if (tuxtxt_cache.vtxtpid == -1 || renderinfo.fb == -1 || renderinfo.sx == -1 || renderinfo.ex == -1 || renderinfo.sy == -1 || renderinfo.ey == -1)
 	{
 		printf("TuxTxt <Invalid Param(s)>\n");
-		return;
+		return -1;
 	}
 
 	/* initialisations */
@@ -274,7 +274,7 @@ int tuxtxt_run_ui(int pid, int demux)
 			tuxtxt_close();
 		}
 #endif
-		return;
+		return 0;
 	}
 	
 	/* main loop */
-- 
2.17.1

